src/lib/Microsoft.Health.Common/Auth/Provider/AzureCredentialProvider.cs

```diff
@@ -0,0 +1,38 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using Azure.Identity;
+
+namespace Microsoft.Health.Common.Auth
+{
+    public class AzureCredentialProvider : IAzureCredentialProvider
+    {
+        private CredentialOptions _credentialOptions;
+
+        public AzureCredentialProvider(ICredentialOptions options)
```

Comment:
>ICredentialOptions [](start = 39, length = 18)

Per my comment on the interface I would be in favor of just going with the credential class without the interface.